src/components/Card.js

```diff
@@ -1,51 +1,87 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { name, link, _id, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick,
+    handleDislikeClick
+  ) {
     this._name = name;
     this._link = link;
+    this._id = _id;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._isLiked = isLiked;
+    this._cardElement = document
+      .querySelector(this._cardSelector)
+      .content.querySelector(".card")
+      .cloneNode(true);
+    this._handleLikeClick = handleLikeClick;
+    this._handleDislikeClick = handleDislikeClick;
   }
 
-  //DECONSTRUCTING SETEVENTLISTENERS
-  _toggleLike = () => {
-    this._likeButton.classList.toggle("card__icon-button_active");
-  };
-
   _setEventListeners() {
     //LIKE BUTTON
-    this._likeButton.addEventListener("click", this._toggleLike);
-
+    this._likeButton.addEventListener("click", () => {
+      if (this._isLiked) {
+        this._handleDislikeClick(this._id, this);
+      } else {
+        this._handleLikeClick(this._id, this);
+      }
+    });
     //DELETE BUTTON
-    this._deleteButton.addEventListener("click", this._deleteCard);
+    this._deleteButton.addEventListener("click", () =>
+      this._handleDeleteClick(this)
+    );
     //IMAGE BUTTON
     this._cardImage.addEventListener("click", () => {
       this._handleImageClick(this._name, this._link);
     });
   }
 
-  _deleteCard = (e) => {
+  deleteCard() {
     this._cardElement.remove();
-    this._cardElement = null;
-  };
+  }
 
   getView() {
-    this._cardElement = document
-      .querySelector(this._cardSelector)
-      .content.querySelector(".card")
-      .cloneNode(true);
-
     this._likeButton = this._cardElement.querySelector(".card__icon");
     this._deleteButton = this._cardElement.querySelector(
       ".card__delete-button"
     );
+
     this._cardImage = this._cardElement.querySelector(".card__image");
     this._cardTitle = this._cardElement.querySelector(".card__title");
 
     this._cardImage.src = this._link;
     this._cardImage.alt = this._name;
     this._cardTitle.textContent = this._name;
 
+    if (this._isLiked) {
+      this.renderLike();
+    }
+
     this._setEventListeners();
     return this._cardElement;
   }
+
+  like() {
+    this._isLiked = !this._isLiked;
+    this._cardElement
+      .querySelector(".card__icon")
```

Comment:
You've found `this._likeButton` in line 49. Please, use it everywhere

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.